### PR TITLE
feat(common): add GrpcChannelArgumentsNativeOption

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #6307 

Add a common option to give users full control over their gRPC channel configuration.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7194)

### CODE REVIEW

File: google/cloud/grpc_options.h [hunk 1]
```diff
@@ -50,7 +50,13 @@ struct GrpcNumChannelsOption {
  *
  * This option gives users the ability to set various arguments for the
  * underlying `grpc::ChannelArguments` objects that will be created. See the
- * gRPC documentation for more details about available options.
+ * gRPC documentation for more details about available channel arguments.
+ *
+ * Our library will always start with the native object from
```
Comment (coryan) [inline google/cloud/grpc_options.h:55]: Consider `@note` or `@warning`

**Review by coryan — APPROVED** [review]
LGTM. Some (optional) changes to the documentation.

This does not use the argument yet, right?

File: google/cloud/grpc_options.h (hunk 1 above)
Comment (dbolduc) [reply google/cloud/grpc_options.h]: Done.

**Comment by dbolduc** [comment]
The option is now in use by every library except Bigtable (which does not yet call `MakeChannelArguments`). See the one-line change in `grpc_options.cc`. (The design doc had a 4 line change, but I realized three of the lines were unnecessary).

**Pull request merged**